### PR TITLE
AESinkAUDIOTrack: Adjustments for AML

### PR DESCRIPTION
**Pull request by fritsch**
This brings AMLogic back in line with the version around by wetek, minix and others. I ported the code to v17 with extraordinary support by @koying. @ChristianTroy helped testing and provided valuable feedback.

### CODE REVIEW

**Comment by fritsch** [comment]
@FernetMenta please have a look at little ActiveAE change, please.

File: xbmc/platform/android/jni/AudioFormat.cpp [hunk 1]
```diff
@@ -94,6 +94,19 @@ void CJNIAudioFormat::PopulateStaticFields()
...
+        int value = -1;
+        GetStaticValue(c, value, "ENCODING_DTSHD");
+        if (value != -1)
+          CJNIAudioFormat::ENCODING_DTS_HD = value;
+        GetStaticValue(c, value, "ENCODING_DTSHD_MA");
+        if (value != -1)
+          CJNIAudioFormat::ENCODING_DTS_HD = value;
+
+        GetStaticValue(c, value, "ENCODING_TRUEHD");
+          if (value != -1)
```
Comment (Jalle19) [inline xbmc/platform/android/jni/AudioFormat.cpp]: Too much indentation

**Comment by fritsch** [comment]
@koying we want to try that? All fine with you?

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**